### ISSUE TITLE
Enum not iterable when accessed through __class__

### ISSUE THREAD

**Issue by steverice** [microsoft/pyright#1880]
**Describe the bug**
Pyright incorrectly thinks that an `Enum` subclass is not iterable when accessed through the `__class__` attribute of an instance, but correctly interprets the class as iterable when accessed by name or a classmethod.

**To Reproduce**
```python
from enum import Enum

class MyEnum(Enum):
    a = 1
    b = 1

    @classmethod
    def clsfunc(cls):
        reveal_type(cls)  # Type[MyEnum]
        return [x for x in cls]

    def instancefunc(self):
        reveal_type(self.__class__)  # Type[MyEnum]
        return [x for x in self.__class__]  # error: "Type[MyEnum]" is not iterable

def func():
    reveal_type(MyEnum)  # Type[MyEnum]
    return [x for x in MyEnum]

```

**Expected behavior**
Pyright should not report an error.

**VS Code extension or command-line**
`pyright` CLI version 1.1.140

**Additional context**
https://github.com/microsoft/pyright/issues/1050 may be related.

The bug doesn't seem to be specific to the type annotations on `Enum`, as this equivalent setup produces the same results:
```python
class MyMeta(type):
    def __iter__(cls):
        return (x for x in [1, 2])

class MyClass(metaclass=MyMeta):
    @classmethod
    def clsfunc(cls):
        return [x for x in cls]

    def instancefunc(self):
        return [x for x in self.__class__]  # error: "Type[MyClass]" is not iterable

```

**Comment by erictraut** [comment]
Thanks for the bug report. This will be fixed in the next release of pyright.

**Comment by erictraut** [comment]
This is fixed in pyright 1.1.141, which I just published. It will also be included in the next release of pylance.
